sources/api/apiserver/src/server/ephemeral_storage.rs

```diff
@@ -269,18 +282,28 @@ pub fn ephemeral_devices() -> Result<Vec<String>> {
 }
 
 /// allowed_bind_dirs returns a set of the directories that can be bound to ephemeral storage, which
-/// varies based on the variant
-pub fn allowed_bind_dirs(variant: &str) -> HashSet<&'static str> {
-    let mut allowed = HashSet::from(["/var/lib/containerd", "/var/lib/host-containerd"]);
+/// varies based on the variant, a set of the prefixes of directories that are allowed to be bound.
+/// and a set of substrings that are disallowed in the directory name.
+pub fn allowed_bind_dirs(
+    variant: &str,
+) ->  BindDirs{
+    let mut allowed_exact = HashSet::from(["/var/lib/containerd", "/var/lib/host-containerd"]);
     if variant.contains("k8s") {
-        allowed.insert("/var/lib/kubelet");
-        allowed.insert("/var/log/pods");
+        allowed_exact.insert("/var/lib/kubelet");
+        allowed_exact.insert("/var/log/pods");
     }
     if variant.contains("ecs") {
-        allowed.insert("/var/lib/docker");
-        allowed.insert("/var/log/ecs");
+        allowed_exact.insert("/var/lib/docker");
+        allowed_exact.insert("/var/log/ecs");
+    }
+    let allowed_prefixes = HashSet::from(["/mnt/"]);
+    let disallowed_contains: = HashSet::from(["..", "/mnt/.ephemeral"]);
```

Comment:
Looks like we have a typo and it either wants a type annotation here or to remove the `:`: https://github.com/bottlerocket-os/bottlerocket-core-kit/actions/runs/12640012198/job/35262690972?pr=320
```suggestion
    let disallowed_contains = HashSet::from(["..", "/mnt/.ephemeral"]);
```

Reply:
thanks @yeazelm , that was a silly thing on my part...should be fixed now!

Reply:
Thanks! That looks better!